MapboxVisionAR/Render/ARScene/ConcreteNodes/ARLaneNode/ARLaneNode.swift

```diff
@@ -0,0 +1,103 @@
+import MetalKit
+import simd
+
+/// Represents AR Lane that can be rendered as a part of `ARScene`.
+class ARLaneNode: ARNode {
+    // MARK: - Properties
+
+    /// Attributes that define the appearance of AR lane.
+    private(set) var arMaterial = ARMaterial()
+    /// Width of AR lane in meters.
+    private(set) var width: Float = 1.0
+
+    // MARK: - Lifecycle
+
+    /**
+     Creates an instance of `ARLaneNode`.
+     The instance has `lane` node type.
+     */
+    init() {
+        super.init(with: .lane)
+    }
+
+    // MARK: - Public methods
+
+    /**
+     Set a new color for underlying AR Lane.
+
+     Method does nothing if `laneColor` does not have compatible color space
+     or there's no underlying AR lane's representation.
+
+     - Parameters:
+       - laneColor: New color of AR lane. It must be in RGBA color space.
+     */
+    func set(laneColor: UIColor) {
+        if let (red, green, blue, alpha) = laneColor.rgbaComponents() {
+            let newARLaneColor = float4(Float(red),
+                                        Float(green),
+                                        Float(blue),
+                                        Float(alpha))
+            self.arMaterial.diffuseColor = newARLaneColor
+        }
+    }
+
+    /**
+     Set a new width for AR lane.
+
+     Method does nothing if there's no underlying AR lane's representation.
+
+     - Parameters:
+       - laneWidth: Width of AR lane in meters.
+     */
+    func set(laneWidth: Float) {
+        self.width = laneWidth
+    }
+
+    /**
+     Set new position of light source for AR lane.
+
+     Method does nothing if there's no underlying AR lane's representation.
+
+     - Parameters:
+       - lightPosition: Position of a light source for AR lane.
+     */
+    func set(lightPosition: float3) {
+        self.arMaterial.light?.position = lightPosition
+    }
+
+    /**
+     Set a new color of a light source for AR lane.
+
+     Method does nothing if `laneLightColor` does not have compatible color space
+     or there's no underlying AR lane's representation.
+
+     - Parameters:
+       - laneLightColor: Color of a light source for AR lane. It must be in RGB color space.
+     */
+    func set(laneLightColor: UIColor) {
+        if let (red, green, blue) = laneLightColor.rgbComponents() {
+            let newLaneLightColor = float3(Float(red),
+                                           Float(green),
+                                           Float(blue))
+            self.arMaterial.light?.color = newLaneLightColor
```

Comment:
haven't found the place where we set a value for `arMaterial.light`, seems like it is `nil` at the init and it can't change its value to any other because we always use soft unwrapping like in this line